ibis/backends/base/sql/alchemy/__init__.py

```diff
@@ -698,6 +705,31 @@ def insert(
                 f"The given obj is of type {type(obj).__name__} ."
             )
 
+    def _compile_udfs(self, expr) -> Iterator[str]:
+        for udf_node in an.find_scalar_udfs(expr.op()):
+
+            @self.add_operation(type(udf_node))
```

Comment:
Could we avoid doing `add_operation` for each of these types, and instead rely on a single handler on the `ScalarUDF` base class registered statically in the backend that handles the operation, possibly exactly as you have below? Having a side effect in `_compile_udfs` feels wrong to me.

Reply:
Yeah, that should work!